### DISCUSSION TITLE
아코, 바론, 우코, 코난, 베로의 github 연동기

### DISCUSSION THREAD

**Discussion by Cyma-s** [woowacourse-teams/2023-shook#7] [General]
# slack에 github 연동하기

현재 팀 프로젝트에서 주 메신저를 slack을 사용합니다. 프로젝트 github repository에서 발생한 알림을 slack에 연동하는 방법이 필요했습니다. 따라서 프로젝트 slack 채널에 github을 연동하고자 했습니다. 

1. slack의 더보기에서 앱 카테고리에 들어갑니다.
<img width="400" alt="slack연동1" src="https://github.com/woowacourse-teams/2023-shook/assets/70303795/0c338db5-8485-404f-b96f-050d1f19dd39">

2. slack에 github를 추가합니다.

<img width="400" alt="slack연동2" src="https://github.com/woowacourse-teams/2023-shook/assets/70303795/ca7a4d32-3274-4bf9-9a68-4e55cfcc4592">

3. 알림을 받을 채널의 세부정보에 들어갑니다.
<img width="263" alt="slack연동3" src="https://github.com/woowacourse-teams/2023-shook/assets/70303795/56e8c0ef-c064-441d-8cf9-55a8d1e7b3be">

3-1. 통합 탭에 들어가서 github를 추가합니다.

<img width="400" alt="slack연동4" src="https://github.com/woowacourse-teams/2023-shook/assets/70303795/24e947bf-2f22-49c3-af47-ce0456c06590">

채널에 등록하면 다음과 같은 알림을 받을 수 있습니다.

<img width="400" alt="slack연동5" src="https://github.com/woowacourse-teams/2023-shook/assets/70303795/1b32a995-9ca3-4468-a23e-1158690c6628">

4. slack에 repository 연동하기

바로 팀 프로젝트 repository를 등록하고 싶었지만 로그인 과정을 거쳐야 합니다.

```
/github signin 
```

Connect GitHub account를 통해 github와 slack과 연동하면 됩니다.
<img width="400" alt="slack연동6" src="https://github.com/woowacourse-teams/2023-shook/assets/70303795/4dfab178-907d-4ab1-93d2-1998397f77ca">

<img width="400" alt="slack연동7" src="https://github.com/woowacourse-teams/2023-shook/assets/70303795/b53eb2b6-c253-46e6-97a5-9e75a1c96b10">

연동을 시도하면 `enter code`에 인증 코드를 입력해야합니다.

이후에 repository를 등록하려고 하면 다음과 같은 문제점이 발생할 수 있는데 github를 설치하면 됩니다.
<img width="400" alt="slack연동8" src="https://github.com/woowacourse-teams/2023-shook/assets/70303795/a60fb526-1855-4f6c-80f8-88cbdac5ce68">

위를 진행하면 github에 slack을 설치하는 창이 나타납니다.

여기서 우리가 등록할 repository에 설치해주면 됩니다.
<img width="400" alt="slack연동9" src="https://github.com/woowacourse-teams/2023-shook/assets/70303795/27561ab7-6eeb-4c5c-aabe-7c7222302207">

여기에 등록한 repository를 설정한 이후에 slack으로 돌아와서 다음과 같이 명령어를 입력하면 slack과 해당 repository가 연동됩니다.

```
/github subscribe woowacourse-teams/2023-shook
```
<img width="400" alt="slack연동10" src="https://github.com/woowacourse-teams/2023-shook/assets/70303795/fd9ab141-7cfc-41ca-a0ea-ec9967915aa7">

이와 같이 해도 바로 정보를 받을 수 있는 것은 아닙니다. 다음과 같이 명령어를 입력해주어야 issue나 discussion를 받을 수 있습니다.

```
/github subscribe woowacourse-teams/2023-shook [원하는 것 ex) discussions]
```

그러면 다음과 같이 discussions이 생겼을 때 알림을 받을 수 있습니다.
<img width="400" alt="slack연동11" src="https://github.com/woowacourse-teams/2023-shook/assets/70303795/29e4f7d9-8b3f-47da-8cea-ccf8e82dd8e6">

지금까지 github와 slack을 연동해보는 과정을 거쳐보았습니다. 팀 프로젝트에 맞게 slack channel에 커스텀하시면 될 것 같습니다.

### 참고자료

> https://adjh54.tistory.com/5
https://github.com/integrations/slack#configuration

**Comment by splitCoding** [comment]
왜 5명이서함

**Reply by seokhwan-an** [reply]
내가 하면서 많이 헤매서 같이 진행했어

**Comment by Creative-Lee** [comment]
너무 고생많았습니다😄
